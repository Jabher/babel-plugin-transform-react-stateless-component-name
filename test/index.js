import { transformFileSync } from 'babel-core'
import path from 'path'
import plugin from '../src'
import test from 'ava'

const getFixture = ( dir, file = 'App.js' ) =>
  path.join(__dirname, 'fixtures', dir, file)

const snapshotMacro = ( t, fixtureDir, fixtureFile ) => {
  const fixture = getFixture(fixtureDir, fixtureFile)
  const plugins = [ plugin ]
  const presets = [ 'es2015', 'react' ]
  const options = { plugins, presets }
  const { code } = transformFileSync(fixture, options)

  t.snapshot(code)
}

test(
  'ignores default function block when not returning JSX',
  snapshotMacro,
  'export-default-with-function-block-no-jsx',
)

test(
  'ignores default function block when no return statement',
  snapshotMacro,
  'export-default-function-block-without-return',
)

test(
  'default export is a function block returning JSX',
  snapshotMacro,
  'export-default-with-function-block',
)

test(
  'ignores default function when not returning JSX',
  snapshotMacro,
  'export-default-no-jsx',
)

test(
  'sets function name for default export based on filename',
  snapshotMacro,
  'export-default',
)

test(
  'sets displayName for assignment default export',
  snapshotMacro,
  'export-default-assignment',
)

test(
  'sets default export from folder name when index.js',
  snapshotMacro,
  'export-default-from-indexjs',
  'index.js',
)

test(
  'sets displayName for assigned functional component',
  snapshotMacro,
  'assigned-component',
)

test(
  'sets displayName for assigned functional component with block',
  snapshotMacro,
  'assigned-component-block',
)

test(
  'sets displayName for named export functional component',
  snapshotMacro,
  'named-export',
)

test(
  'sets displayName for named export functional component with block',
  snapshotMacro,
  'named-export-block',
)

test(
  'ignores when jsx is assigned to a variable and not a component/function',
  snapshotMacro,
  'ignore-assigned-jsx',
)

test(
  'ignores when displayName is already set on function',
  snapshotMacro,
  'ignore-existing-displayname',
)

test(
  'ignores when displayName is already set on default export function',
  snapshotMacro,
  'export-default-ignore-existing-displayname',
)
