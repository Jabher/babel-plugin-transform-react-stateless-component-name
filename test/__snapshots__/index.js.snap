exports[`default export is a function block returning JSX 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  var message = \'Hello!\';

  return _react2.default.createElement(
    \'h1\',
    null,
    message
  );
};

App.displayName = \'App\';"
`;

exports[`ignores default function block when no return statement 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

exports.default = function () {
  // Do some stuff, but don\'t return anything.
  var message = \'Hello!\';
  var answer = 42;
};"
`;

exports[`ignores default function block when not returning JSX 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

exports.default = function () {
  return new Promise(function (resolve) {
    window.addEventListener(\'scroll\', function () {
      resolve(window.scroll);
    });
  });
};"
`;

exports[`ignores default function block when not returning JSX 2`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  var message = \'Hello!\';

  return message;
};"
`;

exports[`ignores default function when not returning JSX 1`] = `
"\"use strict\";

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

exports.default = function () {
  return 42;
};"
`;

exports[`ignores displayName when if/else block has no return statement 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (_ref) {
  var renderA = _ref.renderA,
      renderB = _ref.renderB;

  var loading = true;

  if (loading) renderA();else {
    renderB();
  }
};"
`;

exports[`ignores when displayName is already set on default export function 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

// We know this will become App from the filename


App.displayName = \'App\';"
`;

exports[`ignores when displayName is already set on function 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelloWorld = function HelloWorld() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

HelloWorld.displayName = \'HelloWorld\';

exports.default = HelloWorld;"
`;

exports[`ignores when jsx is assigned to a variable and not a component/function 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelloWorld = _react2.default.createElement(
  \'h1\',
  null,
  \'Hello!\'
);

exports.default = HelloWorld;"
`;

exports[`sets default export from folder name when index.js 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function exportDefaultFromIndexjs() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

exportDefaultFromIndexjs.displayName = \'exportDefaultFromIndexjs\';"
`;

exports[`sets default export function name with deeply nested JSX return statement 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  if (true) {
    if (true) {
      if (true) {
        if (true) {
          return _react2.default.createElement(
            \'h1\',
            null,
            \'Hello!\'
          );
        }
      }
    }
  }
};

App.displayName = \'App\';"
`;

exports[`sets displayName for assigned functional component 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelloWorld = function HelloWorld() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

HelloWorld.displayName = \'HelloWorld\';
exports.default = HelloWorld;"
`;

exports[`sets displayName for assigned functional component with block 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelloWorld = function HelloWorld() {
  var message = \'Hello!\';

  return _react2.default.createElement(
    \'h1\',
    null,
    message
  );
};

HelloWorld.displayName = \'HelloWorld\';
exports.default = HelloWorld;"
`;

exports[`sets displayName for assignment default export 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var App = function App() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

App.displayName = \'App\';
exports.default = App;"
`;

exports[`sets displayName for named export functional component 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});
exports.HelloWorld = undefined;

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelloWorld = exports.HelloWorld = function HelloWorld() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};
HelloWorld.displayName = \'HelloWorld\';"
`;

exports[`sets displayName for named export functional component with block 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});
exports.HelloWorld = undefined;

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelloWorld = exports.HelloWorld = function HelloWorld() {
  var message = \'Hello!\';

  return _react2.default.createElement(
    \'h1\',
    null,
    message
  );
};
HelloWorld.displayName = \'HelloWorld\';"
`;

exports[`sets displayName when if/else block returns JSX 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  var loading = true;

  if (loading) {
    return _react2.default.createElement(
      \'h1\',
      null,
      \'Loading...\'
    );
  } else {
    return _react2.default.createElement(
      \'h1\',
      null,
      \'Hello!\'
    );
  }
};

App.displayName = \'App\';"
`;

exports[`sets displayName when if/else returns JSX 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  var loading = true;

  if (loading) return _react2.default.createElement(
    \'h1\',
    null,
    \'Loading...\'
  );else return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

App.displayName = \'App\';"
`;

exports[`sets function name for default export based on filename 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  return _react2.default.createElement(
    \'h1\',
    null,
    \'Hello!\'
  );
};

App.displayName = \'App\';"
`;

exports[`works when default export is a function block returning JSX 1`] = `
"'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function App() {
  var message = 'Hello!';

  return _react2.default.createElement(
    'h1',
    null,
    message
  );
};"
`;

exports[`works with preact 1`] = `
"'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Header = require('./Header');

var _Header2 = _interopRequireDefault(_Header);

var _preact = require('preact');

var _preact2 = _interopRequireDefault(_preact);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @jsx preact.h */

var App = function App() {
  return _preact2.default.h(
    'main',
    null,
    _preact2.default.h(_Header2.default, { title: 'Hello!' }),
    _preact2.default.h(
      'h2',
      null,
      'World!'
    )
  );
};

exports.default = App;"
`;
